test/unit-tests/sexp.mlt

```diff
@@ -140,7 +140,7 @@ parse {|"$bar%foo%"|}
 
 parse {|\%{foo}|}
 [%%expect{|
-- : parse_result = Same (Ok [\%{foo}])
+Exception: Invalid_argument "'\\%{foo}' is not a valid dune atom".
```

Comment:
Could you also modify `parse` so that `Invalid_argument` exceptions are captured? So that we can see if the behavior is different or not.